src/app/upgrade.cpp

```diff
@@ -82,3 +82,34 @@ bool upgrade(const bool /*ask*/)
     exportWebUIHttpsFiles();
     return true;
 }
+
+void handleChangedDefaults(const DefaultPreferencesMode mode)
+{
+    struct DefaultValue
+    {
+        QVariant legacy;
+        QVariant current;
+
+        QVariant operator[](const DefaultPreferencesMode mode) const
+        {
+            switch (mode)
+            {
+            case DefaultPreferencesMode::Legacy:
+                return legacy;
+            case DefaultPreferencesMode::Current:
+                return current;
+            default:
```

Comment:
Since you exhaust all possible values, does `default` have any meaning?

Reply:
Unfortunately, C++ enums aren't protected from being assigned with invalid value...